File: Permutation in String.py
# Day :18

from collections import Counter
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        from collections import Counter

        l1 = Counter(s1)
        ans=[]
        prev = 0
        for i in range(len(s1) , len(s2)+1):
            l2 = Counter(s2[prev:i])
            if (l2==l1):
                #ans.append(prev)
                return True
            prev +=1
            l2.clear()
        return False
        

File: PermutationInString.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        int n=s1.size();
        vector<int>charcount(26,0);
        for(char c:s1)
        {
            charcount[c-'a']++;
        }
        vector<int>result;
        vector<int>count(26,0);
        for(int i=0;i<s2.length();i++)
        {
            count[s2[i]-'a']++;
            if(i>=n)
            {
                count[s2[i-n]-'a']--;
            }
            if(count==charcount)
                return true;
        }
        
        return false;
    }
};
